### PR TITLE
Fix#1189- Delete Button Added In Description Alert

### LINKED ISSUE #1189: Implementing Delete functionality in description of Image.
**Actual Behaviour**

There is no Toast msg after the description is updated and no delete option to delete a description of an image.

**Expected Behaviour**

1.Added a delete option in AlertDialog Box
2.Shows a toast msg after every description saved or being deleted
3.The delete button will show error msg by toast if the user tries to delete a description of the image without giving a prior description.
4.EditText with empty string will not be saved as description

**Steps to reproduce it**

**Screenshots of the issue**

20170922-013914 = 250x250

**Would you like to work on the issue?**

Yes ! @anantprsd5 @mohitmanuja @Subhankar29

### PR DESCRIPTION
**Pull request by codedsun**
Fix #1189 Delete Button Added in Description Dialog box for an Image

Changes: Added a Delete Button for alert dialog box of description.Shows Snackbar for any changes, error.

Screenshots for the change: 
Included In the issue

@anantprsd5 Kindly Review!

### CODE REVIEW

File: app/src/main/res/layout/activity_pager.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <RelativeLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
+        android:id="@+id/suneeti"
```
Comment (mohitmanuja) [inline app/src/main/res/layout/activity_pager.xml]: suneeti ? Please user proper name.

Comment (codedsun) [reply app/src/main/res/layout/activity_pager.xml]: @mohitmanuja that was not added ! I pushed the code for the mentors to check as the Snackbar is still not visible. I will be pushing again with changes and correctness

Comment (codedsun) [reply app/src/main/res/layout/activity_pager.xml]: @mohitmanuja chck the issue ! I showed toast first then as Shubhankar said , I showed the snackbar

Comment (anantprsd5) [reply app/src/main/res/layout/activity_pager.xml]: @codesun Snackbar is visible on my device. Just hide the delete button when there is nothing to delete and use proper name for id.

**Comment by codedsun** [comment]
Kindly some other mentor plz review this,it's not working on my devices(Asus and Moto) while working on @anantprsd5 device..
@mohitmanuja @Subhankar29 @vinaysajjanapu

**Comment by codedsun** [comment]
@anantprsd5  @mohitmanuja Merge this.

**Comment by anantprsd5** [comment]
@codedsun Resolve conflicts.

**Comment by codedsun** [comment]
@anantprsd5 Resolved !

**Comment by codedsun** [comment]
@anantprsd5 @mohitmanuja why is the travis ci build failing?

**Comment by codedsun** [comment]
@anantprsd5 @mohitmanuja Merge this :)

**Comment by mohitmanuja** [comment]
Please fix error https://travis-ci.org/fossasia/phimpme-android/builds/290114285#L6381  .

**Comment by codedsun** [comment]
@mohitmanuja fixing soon!

**Comment by codedsun** [comment]
@mariobehling merge this!

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java [hunk 2]
```diff
@@ -690,22 +690,63 @@ public void onClick(DialogInterface dialog, int which) {
 
                     }
                 });
+                descriptionDialogBuilder.setNeutralButton(getString(R.string.delete).toUpperCase(), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                       //Hii
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: Remove unwanted comments

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java [hunk 3]
```diff
@@ -690,22 +690,63 @@ public void onClick(DialogInterface dialog, int which) {
...
+                    //Showing delete button if their's a description available
+                    descriptionDialog.getButton(AlertDialog.BUTTON_NEUTRAL).setEnabled(true);
+                }
+                descriptionDialog.getButton(AlertDialog.BUTTON_NEUTRAL).setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        descriptionDialog.dismiss();
+                            databaseHelper.delete(temp);
+                            SnackBarHandler.show(parentView,"Description Deleted");
+                            //Toast.makeText(context, "Description Deleted", Toast.LENGTH_SHORT).show();
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: Here also

Comment (codedsun) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: doing :)

**Comment by codedsun** [comment]
@anantprsd5 Check now !

**Comment by codedsun** [comment]
@aditya98ak  @mohitmanuja @mayank8318 @sridharjajoo Review this!

**Comment by anantprsd5** [comment]
@codedsun Please check the failing travis

**Comment by anantprsd5** [comment]
@codedsun Any updates on this PR?

**Comment by codedsun** [comment]
@anantprsd5 PR Updated!

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java [hunk 4]
```diff
@@ -929,16 +942,30 @@ public void onClick(DialogInterface dialog, int which) {
...
-                
+
                 descriptionDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
                     @Override
                     public void onClick(View v) {
                         descriptionDialog.dismiss();
                         voiceInput = editTextDescription.getText().toString();
                         if (temp == null) {
                             databaseHelper.addImageDesc(new ImageDescModel(pathForDescription, editTextDescription.getText().toString()));
+                            SnackBarHandler.show(parentView,"Description Saved");
```
Comment (anantprsd5) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: Please don't hardcode string like this

Comment (codedsun) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SingleMediaActivity.java]: @anantprsd5  Ok ! adding to strings.xml and updating

**Pull request merged**